src/gravity/gravity_restart.cpp

```diff
@@ -0,0 +1,100 @@
+// Special functions needed to make restart (init=Read_Grid) consistent with
+// running continuously
+
+#include <cstdio>
+
+#ifdef GRAVITY
+  #include "../gravity/grav3D.h"
+  #include "../io/io.h"
+#endif
+
+#ifdef MPI_CHOLLA
+// provides procID
+  #include "../mpi/mpi_routines.h"
+#endif  // MPI_CHOLLA
+
+#ifdef HDF5
+  #include <hdf5.h>
+#endif
+
+void Gravity_Restart_Filename(char* filename, char* dirname, int nfile)
+{
+#ifdef MPI_CHOLLA
+  sprintf(filename, "%s%d_gravity.h5.%d", dirname, nfile, procID);
+#else
+  sprintf(filename, "%s%d_gravity.h5", dirname, nfile);
+#endif
+}
+
+#if defined(GRAVITY) && defined(GRAVITY_RESTART) && defined(HDF5)
```

Comment:
IMO the code within `GRAVITY_RESTART` should just be part of the `GRAVITY` flag. As a user you would expect gravity to run with all the stuff required for a restart and any serious simulation is going to have to have it on.

Reply:
Again, I agree with Bob - I don't like adding new macros that a user has to be aware of and turn on in order to get what one assumes would be the expected behavior.

Reply:
I'm supportive of this. I added the flag so as to not surprise pre-existing users because previously Cholla did not do this. Happy to remove the flag.